src/resin.cpp

```diff
@@ -15,18 +15,26 @@ void Resin::Setup(const char* applicationUUID, const char* ssid, const char* pas
 
     WiFi.setAutoConnect(true);
     WiFi.setAutoReconnect(true);
-    WiFi.begin(_ssid, _password);
+    WiFi.begin(_ssid.c_str(), _password.c_str());
     while (WiFi.status() != WL_CONNECTED) {
         delay(100);
     }
 
     _httpServer = ESP8266WebServer(80);
     _httpUpdater = ESP8266HTTPUpdateServer();
 
-    _httpServer.on("/id", [&](){
+    _httpServer.on("/id", HTTP_GET, [&]() {
         _httpServer.send(200, "text/plain", _applicationUUID);
     });
 
+    _httpServer.on("/env", HTTP_POST, [&]() {
+        DynamicJsonBuffer jsonBuffer;
+        JsonObject& root = jsonBuffer.parseObject(_httpServer.arg(0));
+        // It is up to the user to process the environment variables
```

Comment:
Does the user have to add that processing code here, or are there hooks to keep it in the main file? Sorry if this is a stupid question. I don't see any hooks though.